docs/user-guide/kubectl-overview.md

```diff
@@ -33,160 +33,271 @@ Documentation for other releases can be found at
+`-o=jsonpath-file=<filename>` | Print the fields defined by the [jsonpath](jsonpath.md) expression in the `<filename>` file.
+`-o=name`     | Print only the resource name and nothing else.
+`-o=wide`     | Output in the plain-text format with any additional information. For pods, the node name is included.
+`-o=yaml`     | Output a YAML formatted API object.
+
+##### Example
+
+In this example, the following command outputs the details for a single pod as a YAML formatted object:
+
+`$ kubectl get pod web-pod-13je7 -o=yaml`
+
+Remember: See the [kubectl](kubectl/kubectl.md) reference documentation for details about which output format is supported by each command.
+
+#### Custom columns
+
+To define custom columns and output only the details that you want into a table, you can use the `custom-columns` option. You can choose to define the custom columns inline or use a template file: `-o=custom-columns=<spec>` or `-o=custom-columns-file=<filename>`.
+
+##### Examples
+
+ * Inline:
+
+      ```console
+      $ kubectl get pods <pod-name> -o=custom-columns=NAME:.metadata.name,RSRC:.metadata.resourceVersion
+     ```
+
+ * Template file:
+
+     ```console
+     $ kubectl get pods <pod-name> -o=custom-columns-file=template.txt
+     ```
+
+     where the `template.txt` file contains:
+
+     ```
+      NAME                    RSRC
+      metadata.name           metadata.resourceVersion
+     ```
+
+The result of running either command is:
+
+```console
+NAME           RSRC
+submit-queue   610995
+```
+
+### Sorting list objects
+
+To output objects to a sorted list in your terminal window, you can add the `--sort-by` flag to a supported `kubectl` command. Sort your objects by specifying any numeric or string field with the `--sort-by` flag. To specify a field, use a [jsonpath](jsonpath.md) expression.
+
+#### Syntax
+
+```
+kubectl [command] [TYPE] [NAME] --sort-by=<jsonpath_exp>
+```
 
-- `command` is a certain operation performed on a given resource(s), such as `create`, `get`, `describe`, `delete` etc.
-- `TYPE` is the type of resource(s). Both singular and plural forms are accepted. For example, `node(s)`, `namespace(s)`, `pod(s)`, `replicationcontroller(s)`, `service(s)` etc.
-- `NAME` is the name of resource(s). `TYPE NAME` can be specified as `TYPE name1 name2` or `TYPE/name1 TYPE/name2`. `TYPE NAME` can also be specified by one or more file arguments: `-f file1 -f file2 ...`, [use YAML rather than JSON](config-best-practices.md) since YAML tends to be more user-friendly for config.
-- `flags` are used to provide more control information when running a command. For example, you can use `-s` or `--server` to specify the address and port of the Kubernetes API server. Command line flags override their corresponding default values and environment variables. [Use short flags sparingly, only for the most frequently used options](../devel/kubectl-conventions.md).
+##### Example
```

Comment:
@jlowdermilk Can you please help validate all these kubeclt example descriptions too?